### PR TITLE
address comments on color_dict

### PR DESCRIPTION
**Pull request by liu-ziyang**
# Description
Address comments on renaming color_dict in label layer

## Type of change
- New feature (non-breaking change which adds functionality)

# References
https://github.com/napari/napari/issues/1356

# How has this been tested?

- manually tested the behavior 

## Final checklist:
- [X] My PR is the minimum possible work for the desired functionality
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: napari/view_layers.py [hunk 1]
```diff
@@ -440,6 +441,8 @@ def view_labels(
         Dimension names.
     show : bool, optional
         Whether to show the viewer after instantiation. by default True.
+    color : dict of int to str
+        Custom label to color mapping
```
Comment (sofroniewn) [inline napari/view_layers.py]:
```suggestion
    color : dict of int to str or array
        Custom label to color mapping. Values must be valid color names or RGBA arrays.
```

File: napari/layers/labels/labels.py [hunk 2]
```diff
@@ -108,6 +108,8 @@ class Labels(Image):
 
         In ERASE mode the cursor functions similarly to PAINT mode, but to
         paint with background label, which effectively removes the label.
+    color : dict of int to str
+            Custom label to color mapping
```
Comment (sofroniewn) [inline napari/layers/labels/labels.py]:
```suggestion
    color : dict of int to str or array
        Custom label to color mapping. Values must be valid color names or RGBA arrays.
```

File: napari/components/add_layers_mixin.py [hunk 3]
```diff
@@ -466,6 +466,8 @@ def add_labels(
             the user and if the data is a list of arrays that decrease in shape
             then it will be taken to be multiscale. The first image in the list
             should be the largest.
+        color : dict of int to str
+            Custom label to color mapping
```
Comment (sofroniewn) [inline napari/components/add_layers_mixin.py]:
```suggestion
    color : dict of int to str or array
        Custom label to color mapping. Values must be valid color names or RGBA arrays.
```

**Comment by sofroniewn** [comment]
Tests are failing as doc strings aren't synched. You've added `color` to the Attributes section in `Labels`, but also need to add it to the `Parameters` section (which is used for the doc string comparison). Can you also move the `color` parameter to come after `properties` in both the function signature and the doc strings for all cases where it appears (Labels, add_labels, view_labels) as we've tried to group the layer specific kwargs together. Hopefully that should fix things!!

Also can you address the comment here https://github.com/napari/napari/pull/1339/files#r439786537 and fix that doc string/ parameter name.

If you could also comment on https://github.com/napari/napari/pull/1339/files#r439794714 that would be great too

Thanks!!

**Comment by jni** [comment]
@ziyangczi @sofroniewn I've pushed some minor docstring updates to this branch, which should be good to go after tests pass!

**Review by jni — APPROVED** [review]

**Review by sofroniewn — APPROVED** [review]
LGTM! Will merge now, windows fail is not related to this PR and popping up on all PRs right now

**Comment by liu-ziyang** [comment]
> @ziyangczi @sofroniewn I've pushed some minor docstring updates to this branch, which should be good to go after tests pass!

hey, is there a plugin that helps you catch docstring format issues? Being still new to python docstrings, I wish there is some way auto-detecting my mistakes

**Comment by sofroniewn** [comment]
> hey, is there a plugin that helps you catch docstring format issues? Being still new to python docstrings, I wish there is some way auto-detecting my mistakes

It' benefit a lot from one too! I think I've heard some discussion about this in places, but I'm not sure. Maybe @napari/core-devs we think about adding this to our linting?

**Comment by tlambert03** [comment]
There’s https://pypi.org/project/pydocstyle/
(And a flake8 extension for it)

I have it enabled locally in my IDE, which is an “opt-in” option for devs.  Or we can add it and enforce it in CI too, (it’s enabled at magicgui) but it can definitely get a little “chatty”.

**Comment by kevinyamauchi** [comment]
> It' benefit a lot from one too! I think I've heard some discussion about this in places, but I'm not sure. Maybe @napari/core-devs we think about adding this to our linting?

I think @jni would benefit too because he wouldn't have to catch so many of my improperly-indented docstrings ;)

> There’s https://pypi.org/project/pydocstyle/
(And a flake8 extension for it)

Thanks! I will try it out.

**Comment by tlambert03** [comment]
If you’re using VSCode, there is a handy extension that will give you a docstring snippet based on your signature:

https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring

Supports numpy format.  There’s probably similar extensions for other IDEs

**Pull request merged**